### PR TITLE
texinfo: Fix build on native FreeBSD

### PR DESCRIPTION
**Pull request by rhelmot**
## Description of changes

part of https://github.com/NixOS/nixpkgs/pull/296581

libintl and locale are both required during build on this new, exciting platform.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] x86_64-freebsd
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/misc/texinfo/common.nix [hunk 1]
```diff
@@ -40,7 +40,8 @@ stdenv.mkDerivation {
 
   # ncurses is required to build `makedoc'
   # this feature is introduced by the ./cross-tools-flags.patch
-  NATIVE_TOOLS_CFLAGS = if crossBuildTools then "-I${getDev buildPackages.ncurses}/include" else null;
+  NATIVE_TOOLS_CFLAGS = lib.optionals crossBuildTools [ "-I${getDev buildPackages.ncurses}/include" ]
+  ++ lib.optionals stdenv.isFreeBSD [ "-I${libintl}/include" ];
```
Comment (alyssais) [inline pkgs/development/tools/misc/texinfo/common.nix:44]: How come it doesn't get added to the search path just by being added as an input?

Comment (rhelmot) [reply pkgs/development/tools/misc/texinfo/common.nix:44]: Oh that is very strange. It seems to be working fine without. Maybe this is a leftover from the impure bootstrap? I'll remove it now.

File: pkgs/development/tools/misc/texinfo/common.nix [hunk 2]
```diff
@@ -40,7 +40,7 @@ stdenv.mkDerivation {
 
   # ncurses is required to build `makedoc'
   # this feature is introduced by the ./cross-tools-flags.patch
-  NATIVE_TOOLS_CFLAGS = if crossBuildTools then "-I${getDev buildPackages.ncurses}/include" else null;
+  NATIVE_TOOLS_CFLAGS = lib.optionals crossBuildTools [ "-I${getDev buildPackages.ncurses}/include" ];
```
Comment (alyssais) [inline pkgs/development/tools/misc/texinfo/common.nix:43]: Might as well drop this hunk now, right?

Comment (rhelmot) [reply pkgs/development/tools/misc/texinfo/common.nix:43]: Oh, oops. I didn't realize that's not how it started...

**Review by alyssais — APPROVED** [review]

File: pkgs/development/tools/misc/texinfo/common.nix [hunk 3]
```diff
@@ -63,7 +63,8 @@ stdenv.mkDerivation {
   installFlags = [ "TEXMF=$(out)/texmf-dist" ];
   installTargets = [ "install" "install-tex" ];
 
-  nativeCheckInputs = [ procps ];
+  nativeCheckInputs = [ procps ]
+    ++ optional stdenv.buildPlatform.isFreeBSD [ freebsd.locale ];
```
Comment (annaleeleaves) [inline pkgs/development/tools/misc/texinfo/common.nix:67]: this looks like it should use `optionals` not `optional`

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**